src/foam/u2/crunch/CrunchController.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+foam.CLASS({
+  package: 'foam.u2.crunch',
+  name: 'CrunchController',
+  documentation: `
+    Defines behaviour for invocation of CRUNCH-related views.
+  `,
+
+  implements: [
+    'foam.mlang.Expressions'
+  ],
+
+  imports: [
+    'capabilityDAO',
+    'ctrl',
+    'prerequisiteCapabilityJunctionDAO',
+    'stack',
+    'userCapabilityJunctionDAO'
+  ],
+
+  requires: [
+    'foam.nanos.crunch.Capability',
+    'foam.nanos.crunch.CapabilityCapabilityJunction',
+    'foam.nanos.crunch.CapabilityJunctionStatus',
+    'foam.nanos.crunch.UserCapabilityJunction',
+    'foam.nanos.crunch.ui.CapabilityWizardlet'
+  ],
+  
+  messages: [
+    { name: 'CANNOT_OPEN_GRANTED', message: 'This capability has already been granted to you.' },
+    { name: 'CANNOT_OPEN_PENDING', message: 'This capability is awaiting approval, updates are not permitted at this time.' }
+  ],
+
+  methods: [
+    function getTC(capabilityId) {
+      var tcList = [];
+      var tcRecurse = () => {}; // and we'll do it with this
```

Comment:
Why is this assigned to this function? Why not just assign to the function below?

Reply:
The body of tcRecurse refers to the variable for recursion, so it needs to be declared before being set to the function below. That said, I could have just done `tcRecurse = null` and that has an advantage of throwing in error if it's ever called in that state, so I'll change that.